internal/datastore/post.go

```diff
@@ -10,13 +11,55 @@ import (
 
 func (d *delphisDB) PutPost(ctx context.Context, post model.Post) (*model.Post, error) {
 	logrus.Debug("PutPost::SQL Create")
-	found := model.Post{}
-	if err := d.sql.Create(&post).First(&found, model.Post{ID: post.ID}).Error; err != nil {
-		logrus.WithError(err).Errorf("Failed to create a post")
+	if err := d.initializeStatements(ctx); err != nil {
 		return nil, err
 	}
 
-	return &found, nil
+	_, err := d.putPostStmt.ExecContext(
+		ctx,
+		post.ID,
+		post.CreatedAt,
+		post.UpdatedAt,
+		post.DiscussionID,
+		post.ParticipantID,
+		post.PostContentID,
+		post.QuotedPostID,
+	)
+	if err != nil {
+		return nil, err
```

Comment:
Please continue logging errors at the lowest level when errors occur. Always with syntax:
```
logrus.WithError(err).Errorf(...)
```
This allows us to make sure the error is logged at the lowest level that we can and any callers don't have to worry about logging it.